airflow/models/baseoperator.py

```diff
@@ -666,7 +666,7 @@ def __init__(
             )
         self.weight_rule = weight_rule
         self.resources: Optional[Resources] = Resources(**resources) if resources else None
-        if task_concurrency and not max_active_tis_per_dag:
```

Comment:
This change seems unrelated?

Reply:
Not related but it seems like what we should do and might be ok in this PR